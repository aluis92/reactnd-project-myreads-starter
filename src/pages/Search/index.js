import React from 'react';
import SearchBooksBar from '../../modules/SearchBooksBar';

function Search() {
    return (
        <div className="search-books">
            <SearchBooksBar />
            <div className="search-books-results">
                <ol className="books-grid" />
            </div>
        </div>
    );
}

export default Search;
