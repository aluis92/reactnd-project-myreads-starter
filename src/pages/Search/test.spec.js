import React from 'react';
import {
    shallow,
    mount,
} from 'enzyme';

import { MemoryRouter } from 'react-router-dom';

import Search from './index';

describe('Search component', () => {
    const defaultSearchProps = {
        onSearch: jest.fn(),
        updateBook: jest.fn(),
    };

    it('should render without crashing', () => {
        expect(shallow(
            <Search {...defaultSearchProps} />,
        )).toMatchSnapshot();
    });

    it('should display InfoText if there is no books to show', () => {
        const searchProps = {
            onSearch: jest.fn(),
            updateBook: jest.fn(),
            allBooks: [],
            searchResults: [],
        };

        const search = shallow(
            <Search {...searchProps} />,
        );

        expect(search.find('InfoText').length).toBe(1);
    });

    describe('onSearch method', () => {
        let searchProps;
        let search;

        beforeEach(() => {
            searchProps = {
                onSearch: jest.fn(() => ({
                    then: success => success(),
                })),
                updateBook: jest.fn(),
                allBooks: [{ id: 'a' }],
                searchResults: [{ id: 'a' }],
            };

            search = mount(
                <MemoryRouter>
                    <div>
                        <Search {...searchProps} />,
                    </div>
                </MemoryRouter>,
            );
        });

        it('should change emptyQuery state to false when call onSearch with text string', () => {
            const searchInstance = search.find('Search').instance();

            expect(searchInstance.state.emptyQuery).toBe(true);
            search.find('Search').instance().onSearch('abc');
            expect(searchInstance.state.emptyQuery).toBe(false);
        });

        it('should change emptyQuery state to true when call onSearch with text string', () => {
            const searchInstance = search.find('Search').instance();

            expect(searchInstance.state.emptyQuery).toBe(true);
            search.find('Search').instance().onSearch();
            expect(searchInstance.state.emptyQuery).toBe(true);
        });
    });
});
