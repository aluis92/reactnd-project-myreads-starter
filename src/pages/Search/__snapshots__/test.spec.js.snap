// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render without crashing 1`] = `
<div>
  <SearchBooksBar
    onSearch={[Function]}
  />
  <InfoText>
    Type a title or author name to display books...
  </InfoText>
</div>
`;
